front-end/src/components/TableInfoFilters.js

```diff
@@ -0,0 +1,147 @@
+      borderRadius: 0,
+      '&:hover': {
+        borderColor: `${primaryColor}`
+      },
+      color: 'white',
+      ...(highlight && { boxShadow: `0px 0px 1px 1px ${primaryColor}` })
+    };
+  }
+};
+
+function TableInfoFilters({
+  copy,
+  copyShareLink,
+  precinct,
+  onPrecinctChange,
+  tmm,
+  onTMMChange,
+  articleOptions,
+  articleFilters,
+  onArticleFiltersChange,
+  reelectionToggle,
+  onReelectionToggleChange
+}) {
+  return (
+    <section id="table-info" className="col-sm-4 col-md-3 pr-3">
+      <div id="table-heading" className="mb-3 mt-3 d-flex align-items-center flex-sm-column flex-lg-row flex-row">
+        <h1 className="">{precinct ? `${copy.heading} Precinct ${precinct}` : `${copy.heading} All Precincts`}</h1>
+        <button className="m-3 btn btn-primary" onClick={copyShareLink}>
+          {copy.shareText}
+        </button>
+      </div>
+      <br />
+      <h2>{copy.filterTitle}</h2>
+      {/* Reelection Toggle */}
+      <div className="filter">
+        <div className="custom-control custom-switch">
+          <input
+            type="checkbox"
+            className="custom-control-input"
+            checked={reelectionToggle}
+            onChange={onReelectionToggleChange}
+            id="reelectionToggle"
+          />
+          <label className="custom-control-label" htmlFor="reelectionToggle">
+            {copy.reelectionToggleLabel}
+          </label>
+        </div>
+      </div>
+      {/* Precinct Select */}
+      <div className="filter">
+        <label>
+          {copy.precinctFilterLabel}
+          <a
+            className="filter-helper-link"
+            href={copy.precinctFilterHelperTextLink}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {copy.precinctFilterHelperText}
+          </a>
+        </label>
+        {useMemo(
+          () => (
+            <Select
+              placeholder="Select Precinct..."
+              options={precinctOptions}
+              value={precinctOptions.find(precinctOption => precinctOption.value === precinct)}
+              onChange={onPrecinctChange}
+              styles={{
+                ...customSelectStyles,
+                menu: provided => ({
+                  ...provided,
+                  zIndex: 4
+                })
+              }}
+            />
+          ),
+          [precinct]
+        )}
+      </div>
+      {/* Representative Search */}
+      <div className="filter">
+        <label htmlFor="tmmSearch">{copy.tmmSearchLabel}</label>
+        <input
+          type="text"
+          className="form-control rounded-0"
+          style={{ background: 'white' }}
+          value={tmm}
+          onChange={onTMMChange}
+          id="tmmSearch"
+          placeholder={copy.tmmPlaceholder}
+        />
+      </div>
+      {/* Article Select */}
+      <div className="filter">
+        <label>{copy.articleFilterLabel}</label>
+        {useMemo(
+          () => (
```

Comment:
I don't think I've seen a hook used here in a JS expression mid-template before. What is this used for? Just for my own knowledge

Reply:
Yeah so the useMemo hook tells a component when it needs to re-render. I was seeing a bunch of unnecessary re-renders of the select components when other filter-values they didn't care about were being updated. If we were _really_ good devs, we'd break these out into separate components and do the memoization before export, but I think this is sufficient for now. Hopefully? 😄